code/game/machinery/computer/research.dm

```diff
@@ -59,36 +59,23 @@
 		GLOB.chemical_data.update_credits(cred.credit_value)
 		visible_message(SPAN_NOTICE("[user] inserts [cred] in [src], collecting [cred.credit_value] points from sales."))
 		qdel(cred)
-	//Clearance Updating
+	//Clearance Card Updating
 	if(!istype(B, /obj/item/card/id))
 		return
 	var/obj/item/card/id/silver/clearance_badge/card = B
 	if(!istype(card))
 		visible_message(SPAN_NOTICE("[user] swipes their ID card on \the [src], but it is refused."))
 		return
-	if(card.clearance_access <= GLOB.chemical_data.clearance_level || (card.clearance_access == 6 && GLOB.chemical_data.clearance_level >= 5 && GLOB.chemical_data.clearance_x_access))
-		visible_message(SPAN_NOTICE("[user] swipes the clearance card on [src], but nothing happens."))
-		return
 	if(user.real_name != card.registered_name)
-		visible_message(SPAN_WARNING("WARNING: ILLEGAL CLEARANCE USER DETECTED. CARD DATA HAS BEEN WIPED."))
-		card.clearance_access = 0
+		visible_message(SPAN_WARNING("WARNING: ILLEGAL CLEARANCE USER DETECTED. ABORTING."))
 		return
 
-	var/give_level
-	var/give_x = FALSE
-	if(card.clearance_access == 6)
-		give_level = 5
-		give_x = TRUE
-	else
-		give_level = card.clearance_access
-
-	GLOB.chemical_data.clearance_level = give_level
-	if(give_x)
-		GLOB.chemical_data.clearance_x_access = TRUE
-		GLOB.chemical_data.reached_x_access = TRUE
+	var/credits_to_add = max(card.credits_to_give - GLOB.chemical_data.credits_gained, 0)
+	GLOB.chemical_data.update_credits(credits_to_add)
+	GLOB.chemical_data.credits_gained = max(GLOB.chemical_data.credits_gained, card.credits_to_give)
```

Comment:
```suggestion
	if(credits_to_add)
		GLOB.chemical_data.update_credits(credits_to_add)
		GLOB.chemical_data.credits_gained += credits_to_add
```
Existing way seems needlessly convoluted for what you're trying to accomplish of just limiting the bonus to the best card swiped.